frb_codegen/src/utils.rs

```diff
@@ -89,10 +89,59 @@ impl Display for BlockIndex {
     }
 }
 
-#[extend::ext]
-impl std::path::Path {
+pub trait PathExt {
+    fn file_name_str(&self) -> Option<&str>;
+    fn directory_name_str(&self) -> Option<&str>;
+}
+
+impl PathExt for std::path::Path {
     #[inline]
     fn file_name_str(&self) -> Option<&str> {
-        self.file_name().and_then(OsStr::to_str)
+        self.file_name().and_then(std::ffi::OsStr::to_str)
     }
+    #[inline]
+    fn directory_name_str(&self) -> Option<&str> {
+        self.parent().and_then(|p| p.to_str())
+    }
+}
+
+/// given 2 paths, output the relative path from the 1st one to the 2ed one.
+/// # Examples
+/// ```
+/// use lib_flutter_rust_bridge_codegen::utils::relative_path;
+/// let output_1 = relative_path("./a/b/c", "./a/d/e");
+/// assert_eq!(std::path::PathBuf ::from(output_1), std::path::PathBuf ::from("../../d/e"));
+/// let output_2 = relative_path("./a/d/e", "./a/b/c");
+/// assert_eq!(std::path::PathBuf ::from(output_2), std::path::PathBuf ::from("../../b/c"));
+/// let output_3 = relative_path("./a/b/c", "./a/b/c/d");
+/// assert_eq!(std::path::PathBuf ::from(output_3), std::path::PathBuf ::from("d"));
+/// let output_4 = relative_path("./a/b/c", "./a/b/c/");
+/// assert_eq!(std::path::PathBuf::from(output_4),std::path::PathBuf::from(""));
+/// ```
+pub fn relative_path(from: &str, to: &str) -> String {
```

Comment:
uber-nit: I guess there should already exists some crate that does this, so we do not need to write by hand?

Reply:
I didn't find someone to do so.

Reply:
What about: 

1. https://stackoverflow.com/questions/39340924/given-two-absolute-paths-how-can-i-express-one-of-the-paths-relative-to-the-oth ie https://crates.io/crates/pathdiff
2. https://docs.rs/relative-path/latest/relative_path/struct.RelativePath.html#method.relative ie https://crates.io/crates/relative-path

Reply:
Hmm pathdiff already in the cargo.toml (https://github.com/fzyzcjy/flutter_rust_bridge/blob/864675c5ba5269f0f665045c43cb0818bb9b9f97/frb_codegen/Cargo.toml#L34) so let's use it

Reply:
Fixed in https://github.com/fzyzcjy/flutter_rust_bridge/pull/1024/commits/d72716876f438134dad99b2781564f6beb682819.